[CI] {p0=mixed_cluster/30_ml_jobs_crud/Test get old cluster categorization job} fails

**Build scan**:

* https://gradle-enterprise.elastic.co/s/3ozjbmlaaxyru
* https://gradle-enterprise.elastic.co/s/xssbbmd4njsbe
* https://gradle-enterprise.elastic.co/s/urkh2ymlzxk56

**Repro line**:

```
./gradlew ':x-pack:qa:rolling-upgrade:v7.11.0#bwcTest'
```

**Reproduces locally?**:

No

**Applicable branches**:

master

**Failure history**:

Started failing yesterday: https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-7d,mode:quick,to:now))&_a=(columns:!(_source),filters:!(),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:'UpgradeClusterClientYamlTestSuiteIT%20AND%20%22Test%20get%20old%20cluster%20categorization%20job%22'),sort:!(process.time-start,desc))

However, looking more closely the failures yesterday are not the same problem.  The `[jobs.0.node] doesn't have a true value` error started today.

**Failure excerpt**:

```
java.lang.AssertionError: Failure at [mixed_cluster/30_ml_jobs_crud:77]: field [jobs.0.node] doesn't have a true value
```

--- a/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/process/autodetect/state/ModelSizeStats.java
+++ b/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/process/autodetect/state/ModelSizeStats.java
@@ -226,6 +226,15 @@ public class ModelSizeStats implements ToXContentObject, Writeable {
         totalPartitionFieldCount = in.readVLong();
         bucketAllocationFailuresCount = in.readVLong();
         memoryStatus = MemoryStatus.readFromStream(in);
+        if (in.getVersion().onOrAfter(Version.V_7_11_0)) {
+            if (in.readBoolean()) {
+                assignmentMemoryBasis = AssignmentMemoryBasis.readFromStream(in);
+            } else {
+                assignmentMemoryBasis = null;
+            }
+        } else {
+            assignmentMemoryBasis = null;
+        }
         if (in.getVersion().onOrAfter(Version.V_7_7_0)) {
             categorizedDocCount = in.readVLong();
             totalCategoryCount = in.readVLong();
@@ -246,15 +255,6 @@ public class ModelSizeStats implements ToXContentObject, Writeable {
             deadCategoryCount = 0;
             failedCategoryCount = 0;
             categorizationStatus = CategorizationStatus.OK;
-        }
-        if (in.getVersion().onOrAfter(Version.V_7_11_0)) {
-            if (in.readBoolean()) {
-                assignmentMemoryBasis = AssignmentMemoryBasis.readFromStream(in);
-            } else {
-                assignmentMemoryBasis = null;
-            }
-        } else {
-            assignmentMemoryBasis = null;
         }
         logTime = new Date(in.readVLong());
         timestamp = in.readBoolean() ? new Date(in.readVLong()) : null;
@@ -286,6 +286,14 @@ public class ModelSizeStats implements ToXContentObject, Writeable {
         out.writeVLong(totalPartitionFieldCount);
         out.writeVLong(bucketAllocationFailuresCount);
         memoryStatus.writeTo(out);
+        if (out.getVersion().onOrAfter(Version.V_7_11_0)) {
+            if (assignmentMemoryBasis != null) {
+                out.writeBoolean(true);
+                assignmentMemoryBasis.writeTo(out);
+            } else {
+                out.writeBoolean(false);
+            }
+        }
         if (out.getVersion().onOrAfter(Version.V_7_7_0)) {
             out.writeVLong(categorizedDocCount);
             out.writeVLong(totalCategoryCount);
@@ -296,14 +304,6 @@ public class ModelSizeStats implements ToXContentObject, Writeable {
                 out.writeVLong(failedCategoryCount);
             }
             categorizationStatus.writeTo(out);
-        }
-        if (out.getVersion().onOrAfter(Version.V_7_11_0)) {
-            if (assignmentMemoryBasis != null) {
-                out.writeBoolean(true);
-                assignmentMemoryBasis.writeTo(out);
-            } else {
-                out.writeBoolean(false);
-            }
         }
         out.writeVLong(logTime.getTime());
         boolean hasTimestamp = timestamp != null;

